src/stores/PreferenceStore.ts

```diff
@@ -692,6 +719,10 @@ export class PreferenceStore {
         this.clearPreferences([PreferenceKeys.CATALOG_DISPLAYED_COLUMN_SIZE, PreferenceKeys.CATALOG_TABLE_SEPARATOR_POSITION]);
     };
 
+    @action resetTelemetrySettings = () => {
+        this.clearPreferences([PreferenceKeys.TELEMETRY_CONSENT_SHOWN, PreferenceKeys.TELEMETRY_MODE, PreferenceKeys.TELEMETRY_UUID, PreferenceKeys.TELEMETRY_LOGGING]);
```

Comment:
I went through the code although I'm not familiar with `axios` and `idb`
I have one question about the user id design: do we expect that the user generates a new user ID by clicking the `restore defaults` button?

Reply:
@YuHsuan-Hwang thanks, I have fixed this. It should now generate and display a new user ID

Reply:
@veggiesaurus sorry, I mean if we allow user to easily generate a new ID, will this causes the number of IDs to be much greater than the actual number of users?

Reply:
yes, but I'm not really sure what the alternative is. Not allowing the user to generate a new uuid might be problematic :thinking: I'm trying to see what Atom does

Reply:
perhaps the middle ground is to _not_ generate a new token when you click the "reset defaults" button, and users could always delete the entry in their preferences manually to get a new token if they really wanted to? (or run `app.preferenceStore.clearPreferences(["telemetryUuid"])` as a snippet)

@kswang1029 what do you think?

Reply:
we can emphasize how to reset the ID in in-app help

Reply:
:+1: ok. Or at least in the privacy policy. I think this is definitely more of an "expert user" thing, so I agree with you

Reply:
I've made the required changes, please have a look again